alg/sha256.c

```diff
@@ -187,7 +187,8 @@ usehw(void)
 
 /*
  * SHA256 block compression function.  The 256-bit state is transformed via
- * the 512-bit input block to produce a new state.
+ * the 512-bit input block to produce a new state.  The arrays W and S may be
```

Comment:
This comment might be overkill in `sha256.c`, but it's definitely necessary in the upcoming `SHA256_Transform_sse2()`, and I figured that the functions should have similar comments unless they had to diverge.

(Note that we don't need it in `sha256_shani.c`, because `SHA256_Transform_shani()` doesn't take W and S as paramters.